File: problems/2091_removing_minimum_and_maximum_from_array/solution.py
class Solution:
    def minimumDeletions(self, nums: list[int]) -> int:
        n = len(nums)
        min_idx = nums.index(min(nums))
        max_idx = nums.index(max(nums))
        left = min(min_idx, max_idx) + 1
        right = n - max(min_idx, max_idx)
        both = left + right
        from_left = max(min_idx, max_idx) + 1
        from_right = n - min(min_idx, max_idx)
        return min(both, from_left, from_right)

File: problems/2091_removing_minimum_and_maximum_from_array/Solution.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int n = nums.length, minIdx = 0, maxIdx = 0;
        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[minIdx]) minIdx = i;
            if (nums[i] > nums[maxIdx]) maxIdx = i;
        }
        int left = Math.min(minIdx, maxIdx) + 1;
        int right = n - Math.max(minIdx, maxIdx);
        int fromLeft = Math.max(minIdx, maxIdx) + 1;
        int fromRight = n - Math.min(minIdx, maxIdx);
        return Math.min(left + right, Math.min(fromLeft, fromRight));
    }
}
